source/include/dqm4hep/XMLParser.h

```diff
@@ -149,6 +149,18 @@ namespace dqm4hep {
        *  @brief  Whether to allow the use of environment variable while resolving constants
        */
       bool allowEnvVariables() const;
+      
+      /**
+       *  @brief  Set whether to process XML for loop elements
+       *
+       *  @param  process whether to process for loops
+       */
+      void setProcessForLoops(bool process);
```

Comment:
I am not really following why it would not be always active. It might actually be bothersome if for some reason the parameter is set to `false` inadvertently. (copy paste from a different configuration etc.)
I would in this case at least add a `Warning` statement when you try to resolve the [loops](https://github.com/DQM4HEP/dqm4hep-core/pull/66/files#diff-a433282c6a16406141b58dd6ba4eafa7R89)

Reply:
This is then true for every feature in XML parser. What do you think ?

Reply:
Yes, that's right. I guess having everything set to true by default is enough. 
I still think that adding a `warning` (`debug` would be more suitable maybe to avoid getting polluted by messages when you disable it on purpose) when the setting is disabled.

Reply:
I have added warnings for each method that can de/activate a parsing feature.